icu4j/main/classes/core/src/com/ibm/icu/impl/locale/XLikelySubtags.java

```diff
@@ -189,8 +189,8 @@ public LSR makeMaximizedLsrFrom(ULocale locale) {
         String name = locale.getName();  // Faster than .toLanguageTag().
         if (name.startsWith("@x=")) {
             String tag = locale.toLanguageTag();
-            assert tag.startsWith("x-");
-            // Private use language tag x-subtag-subtag...
+            assert tag.startsWith("und-x-");
+            // Private use language tag und-x-subtag-subtag...
```

Comment:
Good catch!

Actually, while CLDR prepends "und-", BCP 47 allows starting with "x-" which is a different part of its BNF.
I suggest changing all of these comments to
```
// Private use language tag x-subtag-subtag... which CLDR changes to und-x-subtag-subtag...
```
Please also fix the same comment in https://github.com/unicode-org/icu/blob/master/icu4c/source/common/loclikelysubtags.cpp#L323